vignettes/epichains.Rmd

```diff
@@ -27,4 +27,392 @@ knitr::opts_chunk$set(
 )
 ```
 
-WIP
+_epichains_ provides methods to analyse and simulate the size and length
+of branching processes with an arbitrary offspring distribution. These
+can be used, for example, to analyse the distribution of chain sizes
+or length of infectious disease outbreaks, as discussed in @farrington2003 and 
+@blumberg2013.
+
+```{r}
+library("epichains")
+```
+
+## Chain likelihoods
+
+### [`likelihood()`](https://epiverse-trace.github.io/epichains/reference/likelihood.html)
+
+This function calculates the likelihood/loglikelihood of observing a vector of outbreak summaries obtained from transmission chains. "Outbreak summaries" here refer to transmission chain sizes or lengths/durations.
+
+`likelihood()` requires a vector of chain summaries (sizes or lengths),
+`chains`, the corresponding statistic to calculate, `statistic`, the offspring
+distribution, `offspring_dist` and its associated parameters. `offspring_dist`
+is specified as the "base name" of the inbuilt distributions in R, for example,
+"pois", "nbinom", etc. `likelihood()` also requires `nsim_obs`, which is the
+number of simulations to run if the likelihoods do not have a closed-form
+solution and must be simulated. This argument will be explained further in
+the next section.
+
+By default, the result is a log-likelihood but if `log = FALSE`, then
+likelihoods are returned. 
+
+Let's look at the following example where we estimate the log-likelihood of
+observing `chain_sizes`.
+```{r}
+set.seed(121)
+# example of observed chain sizes
+# randomly generate 20 chains of size between 1 to 10
+chain_sizes <- sample(1:10, 20, replace = TRUE)
+chain_sizes
+```
+
+```{r}
+# estimate loglikelihood of the observed chain sizes
+likelihood_eg <- likelihood(
+  chains = chain_sizes,
+  statistic = "size",
+  offspring_dist = "pois",
+  nsim_obs = 100,
+  lambda = 0.5
+)
+# Print the estimate
+likelihood_eg
+```
+
+### Joint and individual log-likelihoods
+
+`likelihood()`, by default, returns the joint log-likelihood. If instead,
+the individual log-likelihoods are required, then the `individual` argument
+must be set to `TRUE`. To return likelihoods instead, set `log = TRUE`.
+```{r}
+set.seed(121)
+# example of observed chain sizes
+# randomly generate 20 chains of size between 1 to 10
+chain_sizes <- sample(1:10, 20, replace = TRUE)
+chain_sizes
+```
+
+```{r}
+# estimate loglikelihood of the observed chain sizes
+likelihood_ind_eg <- likelihood(
+  chains = chain_sizes,
+  statistic = "size",
+  offspring_dist = "pois",
+  nsim_obs = 100,
+  lambda = 0.5,
+  individual = TRUE
+)
+# Print the estimate
+likelihood_ind_eg
+```
+
+### How `likelihood()` works
+
+*epichains* ships with functions for the analytical solutions of some
+transmission chain "size" and "length" distributions. For the size
+distributions, we provide the `poisson`, `negative binomial`, and `gamma-borel`
```

Comment:
if given in backtics should this not be `pois`, `nbinom` and `gborel`? Whereas if we write them out we should perhaps keep them in normal font (same below)?

Reply:
I'll remove the backticks.